src/components/BarChart/BarChart-container.tsx

```diff
@@ -48,7 +48,12 @@ const BarChartContainer = (props: BarChartProps) => {
         <Link to={object.walletHash} key={object.walletHash}>
           <div
             className={getClasses(index)}
-            onClick={() => actions.fetchEthereumTransactionsSummed({walletHash:object.walletHash})}
+            onClick={
+              () => {
+                actions.flushEthereumTransactions();
+                actions.fetchEthereumTransactions({walletHash:object.walletHash, page: 1});
+                actions.fetchEthereumTransactionsSummed({walletHash:object.walletHash});
+            }}
```

Comment:
This doesn't look like a good idea.

I wouldn't use onClick handlers here, completley unnecesary and is hard to think and reason about. This is an introduction to very messy codebase.

A much better solution is to read route params from the component where you need this data, and once it changes compare previous route param value to the current one, if it's different then run this action.